backend/module/eCampLib/src/Types/DateBase.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace eCamp\Lib\Types;
+
+use JsonSerializable;
+
+abstract class DateBase extends \DateTime implements JsonSerializable {
+    protected string $FORMAT;
+
+    public function __toString() {
+        return $this->format($this->FORMAT);
+    }
+
+    /**
+     * formatter used with json_encode.
+     */
+    public function jsonSerialize() {
+        return $this->__toString();
+    }
+
+    /**
+     * @param null|string $time   Date in string representation
+     * @param null|string $format Allowed format of the date
+     *
+     * @throws InvalidDateFormatException
+     * @throws InvalidZoneOffsetException
+     */
+    protected function checkFormat(?string $time, ?string $format): void {
```

Comment:
In case you decide to keep this method here, consider adding default parameter values, so this function can be called as `checkFormat('2020-01-01')` without explicitly specifying `null` for `$format`. Adding default values to parameters on overridden methods in subclasses is allowed in PHP.